src/main/java/org/jabref/gui/sidepane/SidePaneType.java

```diff
@@ -1,10 +1,38 @@
 package org.jabref.gui.sidepane;
 
+import org.jabref.gui.actions.Action;
+import org.jabref.gui.actions.StandardActions;
+import org.jabref.gui.icon.IconTheme;
+import org.jabref.gui.icon.JabRefIcon;
+import org.jabref.logic.l10n.Localization;
+
 /**
  * Definition of all possible components in the side pane.
  */
 public enum SidePaneType {
-    OPEN_OFFICE,
-    WEB_SEARCH,
-    GROUPS
+    OPEN_OFFICE("OpenOffice/LibreOffice", IconTheme.JabRefIcons.FILE_OPENOFFICE, StandardActions.TOOGLE_OO),
+    WEB_SEARCH(Localization.lang("Web search"), IconTheme.JabRefIcons.WWW, StandardActions.TOGGLE_WEB_SEARCH),
+    GROUPS(Localization.lang("Groups"), IconTheme.JabRefIcons.TOGGLE_GROUPS, StandardActions.TOGGLE_GROUPS);
```

Comment:
I'm absolutely no fan of this enum, because I think it breaks the O in the SOLID design principles...
But maybe we need to adress this later as this PR grows too large and since you did not introduce it in the first place, i guess it's ok to leave it for now.

Reply:
Can you elaborate more about how this violates OCP?

Reply:
If I understood it correctly, the programmer should not have to modify existing classes, but just add to them (like another class extending the component) and carry all the information about itself with it instead of relying on another enum.

Right now we have multiple places where we distinguish between the different types of components (often by switch), that would have to be modified, if we would add another sidepanetype (who knows whatever comes someday in our minds).

Reply:
> the programmer should not have to modify existing classes, but just add to them (like another class extending the component)

But then we would have duplication and unnecessary abstraction between `WebSearchComponent` and `OpenOfficeComponent` since we have 2 of 3 components share the same API, I decided to create a standard implementation that most panes will use and an extension opportunity by inheriting `SidePaneComponent`, special implementations can add extra actions to the pane bar by calling `addExtraButtonToHeader()` and handle the new action logic on their own, however, there is a downside for this design that we can't add new API for components because then we would have to cast the component object to the class that has that API, like with `GroupsSidePaneComponent#afterOpening()` although we can get rid of it using binding, so unless we wanna add a new component that introduces a new API we should be fine.

> carry all the information about itself with it instead of relying on another enum.

In my opinion, it's overkill to create 3 classes so that each one implements a method that returns static data, we can create instances for that, inheritance should be used for custom implementation of logic.

> Right now we have multiple places where we distinguish between the different types of components (often by switch), that would have to be modified

We can fix that by adding more properties to `SidePaneComponent` for example adding `visibleProperty`, but then we need to reference `SidePaneComponent` directly to access the visibility of a component, for me, components are internal to `SidePane` and should only be accessed through the interface provided by `SidePane` with passing `SidePaneType` arguments to distinguish the pane.